### PR TITLE
docs: move plugin naming

### LINKED ISSUE #3873: Plugin naming convention should be explained at the beginning of the dev guide
Since people following the plugin dev guide will mostly start creating it at the beginning and name it, I think it makes sense that the first thing we explain is how to name the plugin.

cc @NataliaTepluhina

### PR DESCRIPTION
**Pull request by NataliaTepluhina**
Close #3873 

Moved a plugin naming convention to the beginning of the Plugin dev guide.

### CODE REVIEW

File: docs/dev-guide/plugin-dev.md [hunk 1]
```diff
@@ -34,6 +34,40 @@ So, typical CLI plugin folder structure looks like the following:
 └── ui.js         # Vue UI integration (optional)
 ```
 
+## Naming and discoverability
+
+For a CLI plugin to be usable by other developers, it must be published on [npm](https://www.npmjs.com/) following the name convention `vue-cli-plugin-<name>` or `@scope/vue-cli-plugin-<name>`. Following the name convention allows your plugin to be:
```
Comment (Akryum) [inline docs/dev-guide/plugin-dev.md]:
```suggestion
For a CLI plugin to be usable in a Vue CLI project, it must follow the name convention `vue-cli-plugin-<name>` or `@scope/vue-cli-plugin-<name>`. It allows your plugin to be:
```

File: docs/dev-guide/plugin-dev.md [hunk 2]
```diff
@@ -34,6 +34,40 @@ So, typical CLI plugin folder structure looks like the following:
...
+
+- Discoverable by `@vue/cli-service`;
+- Discoverable by other developers via searching;
+- Installable via `vue add <name>` or `vue invoke <name>`.
+
```
Comment (Akryum) [inline docs/dev-guide/plugin-dev.md]: Maybe add a warning block here explaining that not following the naming convention will result in problems?

Comment (NataliaTepluhina) [reply docs/dev-guide/plugin-dev.md]: Done 😸

**Pull request merged**